Framework/Core/Windows/Devices/Gpio/GpioPinEventListener.cs

```diff
@@ -0,0 +1,89 @@
+using System;
+using System.Collections;
+using Microsoft.SPOT;
+
+namespace Windows.Devices.Gpio
+{
+    internal class GpioPinEvent : BaseEvent
+    {
+        public int PinNumber;
+        public GpioPinEdge Edge;
+    }
+
+    internal class GpioPinEventListener : IEventProcessor, IEventListener
+    {
+        // Construction and destruction
+
+        public GpioPinEventListener()
+        {
+            EventSink.AddEventProcessor(EventCategory.Gpio, this);
+            EventSink.AddEventListener(EventCategory.Gpio, this);
+        }
+
+        // IEventProcessor methods
+
+        public BaseEvent ProcessEvent(uint data1, uint data2, DateTime time)
+        {
+            return new GpioPinEvent
+            {
+                // Data1 is packed by PostManagedEvent, so we need to unpack the high word.
+                PinNumber = (int)((data1 >> 16) & 65535u),

```

Comment:
What is the purpose of the '&' operation? Don't (data1 >> 16) and (data >> 16) & 65535u always have the same value here?